src/app/campaign/store/models/availability.models.ts

```diff
@@ -0,0 +1,41 @@
+export interface AvailabilityParams {
+  inventoryId: string;
+  startDate: Date;
+  endDate: Date;
+  zone: string;
+  flightId: number;
+}
+
+export interface InventoryNumbers {
+  allocated: number;
+  availability: number;
+  actuals: number;
+  allocationPreview?: number;
+}
+
+export interface AvailabilityDay {
+  date: Date;
+  numbers: InventoryNumbers;
+}
+
+export const docToAvailabilityDay = (availability: any): AvailabilityDay => {
+  return { date: new Date(availability.date), numbers: { ...availability } };
+};
+
+export interface Availability {
+  params: AvailabilityParams;
+  days: AvailabilityDay[];
+}
+
+export interface AvailabilityWeeklyRollup {
+  startDate: Date;
+  endDate: Date;
+  numbers: InventoryNumbers;
+  days: AvailabilityDay[];
+}
+
+export interface AvailabilityRollup {
```

Comment:
Explicit rollup data model ++

Reply:
the prior rollup models within the model for this were far too clever and harder to keep track of, really annoyed me